core/mastodon/src/test/java/br/com/orcinus/orca/core/mastodon/instance/requester/authentication/RequesterTestScope.extensions.kt

```diff
@@ -16,26 +16,35 @@
 package br.com.orcinus.orca.core.mastodon.instance.requester.authentication
 
 import br.com.orcinus.orca.core.auth.actor.Actor
+import br.com.orcinus.orca.core.mastodon.instance.requester.ClientResponseProvider
 import br.com.orcinus.orca.core.mastodon.instance.requester.Requester
 import br.com.orcinus.orca.core.mastodon.instance.requester.RequesterTestScope
 import br.com.orcinus.orca.core.mastodon.instance.requester.runRequesterTest
+import io.ktor.client.HttpClient
 import kotlin.contracts.ExperimentalContracts
 import kotlin.contracts.InvocationKind
 import kotlin.contracts.contract
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlinx.coroutines.Job
 
 /**
  * Runs an [AuthenticatedRequester]-focused test.
  *
+ * @param clientResponseProvider Defines how the [HttpClient] will respond to requests.
  * @param onAuthentication Action run whenever the [Actor] is authenticated.
+ * @param context [CoroutineContext] in which [Job]s are launched by default.
  * @param body Operation to be performed with the [Requester].
  */
 @OptIn(ExperimentalContracts::class)
 internal inline fun runAuthenticatedRequesterTest(
+  clientResponseProvider: ClientResponseProvider = ClientResponseProvider.ok,
   crossinline onAuthentication: () -> Unit = {},
+  context: CoroutineContext = EmptyCoroutineContext,
```

Comment:
Same here: although unrelated, test whether the body is run in the given context.

Reply:
And also whether the response produced by the provider is the received one.